### PR TITLE
MBS-7752: Replace referer redirects with ?returnto=

### PR DESCRIPTION
**Pull request by mwiencek**
The referer header is optional and may not be sent by the user's browser (particularly if they've chosen to block it). Specify a `returnto` parameter everywhere we might need one.

`redirect_back` has been improved ~to make the infinite redirect checking (currently handled by `$ignore`) more generalized, and~ to block any attempt at redirecting to an external URL.

### CODE REVIEW

File: root/user/UserProfile.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ import commaOnlyList from '../static/scripts/common/i18n/commaOnlyList';
 import {formatCount, formatPercentage} from '../statistics/utilities';
 import formatUserDate from '../utility/formatUserDate';
 import {canNominate} from '../utility/voting';
+import { returnToCurrentPage } from '../utility/returnUri';
```
Comment (reosarevok) [inline root/user/UserProfile.js:27]: Why the padding? :)

Comment (mwiencek) [reply root/user/UserProfile.js:27]: I use a vscode-js-import plugin to help import stuff and it adds those.

**Comment by mwiencek** [comment]
Finally fixed the tests here, should be ready for review

**Review by reosarevok — COMMENTED** [review]
The code looks fine - any specific tests suggested? I tried it for merges, where it seemed to work well. I guess we can't test the beta unsetting locally in any reasonable way. What other places do you feel we should try? :)

**Comment by mwiencek** [comment]
I should be able to test the beta stuff on one of my development servers (I have an prod/beta setup to switch between). I'll check if MBS-6438 is resolved too.

Otherwise, login/logout redirects, plus adding stuff to collections/subscripts would be useful to check!

**Review by reosarevok — APPROVED** [review]
Tested those few things and they seem to work. Happy with this if you tested the beta stuff locally.

**Review by yvanzo — APPROVED** [review]
:shipit:

**Pull request merged**